File: backup_assistant_cloudant.js
/**
  *
  * main() será executado ao clicar em "Run This Action".
  * 
  * @author     Gilvan Junior
  * @company    Magic
  * @contact    https://www.linkedin.com/in/gilvanjunior/
  *
  */
  
const AssistantV1 = require('ibm-watson/assistant/v1');
var Cloudant = require("@cloudant/cloudant");

function main(params) {
    
    const assistant = new AssistantV1({
      iam_apikey: params.IAM_APIKEY,
      version: params.WDC_VERSION,
      url: params.WDC_URL
    });
    
    const cloudant = Cloudant({
        account: params.CLOUDANT_ACCOUNT,
        password: params.CLOUDANT_PASSWORD
    });
    
    console.log(JSON.stringify(params))

    const versioning_db = cloudant.db.use(params.CLOUDANT_DBNAME);
    
    return new Promise(function(resolve, reject){
        var wdc_params = {
            workspace_id: params.WDC_WORKSPACEID,
            _export: true
        };
        assistant.getWorkspace(wdc_params)
          .then(res => {
            var d = new Date();
            d.setHours(d.getHours() - 3);
            var horario = d.toISOString();
          
            res._id = params.WDC_PREFIX + horario;
              
            versioning_db.insert(res, function (err, body, header) {
                if (err) return reject(err);
                return resolve({ "message": "Backup realizado com sucesso!" });
             });
          })
          .catch(err => {
            console.log(err)
          });
    });
    
}

File: backup_assistant_cloudant.py
#
# main() será executado ao clicar em "Run This Action".
# 
# @author     Gilvan Junior
# @contact    https://www.linkedin.com/in/gilvanjunior/
#

import sys
import json
from watson_developer_cloud import AssistantV1
from cloudant.client import Cloudant
from cloudant.error import CloudantException
from cloudant.result import Result, ResultByKey

def main(dict):
    
    serviceUsername = dict['CLOUDANT_ACCOUNT']
    servicePassword = dict['CLOUDANT_PASSWORD']
    serviceURL = dict['CLOUDANT_URL']
    
    # Use the IBM Cloudant library to create an IBM Cloudant client.
    client = Cloudant(serviceUsername, servicePassword, url=serviceURL)
    
    # Connect to the server
    client.connect()
    
    databaseName = dict['CLOUDANT_DBNAME']
    
    # Create an instance of the database.
    myDatabaseDemo = client.create_database(databaseName)
    
    # Check that the database now exists.
    if myDatabaseDemo.exists():
        print("'{0}' successfully created.\n".format(databaseName))
        
    service = AssistantV1(
        iam_apikey=dict['IAM_APIKEY'],
        version=dict['WDC_VERSION'],
        url=dict['WDC_URL']
    )
    
    response=service.get_workspace(
        workspace_id=dict['WDC_WORKSPACEID'],
        export='true'
    ).get_result()
        
    newDocument = myDatabaseDemo.create_document(response)

    return { 'status': 200, 'message': 'Backup realizado com sucesso!' }
